File: add-two-numbers/add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* dummy=new ListNode(0);
        ListNode* temp=dummy; 
        auto *temp1=l1; auto *temp2=l2; int carry=0;
        while(temp1!=NULL or temp2!=NULL or carry)
        {
            int sum=0;
            if(temp1!=NULL){
                sum+=temp1->val; temp1=temp1->next;
            }
            if(temp2!=NULL){
                sum+=temp2->val; temp2=temp2->next;
            }
            sum=sum+carry;
            carry=sum/10;
            ListNode* node=new ListNode(sum%10);
            temp->next=node;
            temp=temp->next;
        }
        return dummy->next;
    }
};

File: add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy_head=ListNode(0)
        l3=dummy_head
        carry=0
        while(l1!=None or l2!=None):
            if(l1==None): l1_val=0
            else: l1_val=l1.val
            if(l2==None): l2_val=0
            else: l2_val=l2.val
            current_sum=l2_val+l1_val+carry
            carry=(int)(current_sum/10)
            last_digit=current_sum%10
            l3.next=ListNode(last_digit)
            if(l1!=None): l1=l1.next
            if(l2!=None): l2=l2.next
            l3=l3.next
        if(carry>0):
            l3.next=ListNode(carry)
            l3=l3.next
        return dummy_head.next